### PR TITLE
OOP school library: add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by Estherstarr**
This PR contains the implementation of the School Library app.
A school library app to record what books are in the library and who borrows them. The app that you will create will allow you to: add new students or teachers, add new books and save records of who borrowed a given book and when.

Highlights:

- [ ] Project setup with gitflow

- [ ] Linters configuration

- [ ] Descriptive Readme with guide

- [ ] Created a Person class with the following:

 - Instance vars: @ id, @ name, and @ Age.

 - Constructor with name, age, and parent_permission as parameter. name and parent_permission are optional and have default values of "Unknown" and true.

 - Getters for @ id, @ name, and @ Age.

 - Setters for @ name and @ Age.

 - Private method is_of_age? that returns true if @Age is greater or equal to 18 and false otherwise.

 - Public method can_use_services? that returns true if person is of age or if they have permission from parents.

- [ ] Create a Student class with the following:
 - It inherits from Person class.

 - Constructor extends parent's constructor by adding @classroom and a parameter for it.

 - Method play_hooky that returns "¯(ツ)/¯".

- [ ] Create a Teacher class with the following:
- It inherits from Person class.

 - Constructor extends parent's constructor by adding https://github.com/specialization and a parameter for it.

 - Override can_use_services? so it always returns true.

Each class is saved in a separate file.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  attr_accessor :name, :age
+
+  def initialize(age:, name: 'Unknown', parent_permission: true)
+    @age = age
+    @name = name
+    @parent_permission = parent_permission
+  end
+
+  def can_use_services?
+    is_of_age? || @parent_permission
```
Comment (DammyShittu) [inline person.rb]: - [ ] I believe this method should be `of_age?`, not `is_of_age?`. Ensure you change that.

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  end
+
+  private
+
+  def of_age?
+    age >= 18
```
Comment (DammyShittu) [inline person.rb]: - [ ] We need to use the instance variable here `@age`, not `age`.

File: person.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# Defines a person class
+class Person
+  attr_reader :id
+  attr_accessor :name, :age
+
+  def initialize(age:, name: 'Unknown', parent_permission: true)
+    @age = age
+    @name = name
+    @parent_permission = parent_permission
```
Comment (DammyShittu) [inline person.rb]: - [ ] You need to initialize the `@id` instance variable so the person object has an id. For example:

  ```
  def initialize(age:, name: 'Unknown', parent_permission: true)
    @id = Random.rand(1..1000)
    ...
  ```

File: student.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+# Defines a student class
+class Student < Person
+  def initialize(age:, classroom:)
+    super(age: age)
```
Comment (DammyShittu) [inline student.rb]: - [ ] Student class inherits from the Person class, so you must also define the other parameters here, so the properties of a student object can be correctly set.

  ```
  def initialize(age:, classroom:, name: 'Unknown', parent_permission: true)
    super(age: age, name: name, parent_permission: parent_permission)
    ...
  end
  ```

File: teacher.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+# Defines a teacher class
+class Teacher < Person
+  def initialize(age:, specialization:)
+    super(age: age)
```
Comment (DammyShittu) [inline teacher.rb]: - [ ] Kindly implement the feedback for the Student class above here also.

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @Estherstarr,

## Highlights
✔️ Passing linters
✔️ Descriptive PR

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: student.rb (hunk 4 above)
Comment (Estherstarr) [reply student.rb]: Implemented

File: person.rb (hunk 1 above)
Comment (Estherstarr) [reply person.rb]: Implemented

File: person.rb (hunk 2 above)
Comment (Estherstarr) [reply person.rb]: implemented

File: teacher.rb (hunk 5 above)
Comment (Estherstarr) [reply teacher.rb]: Implemented

**Review by DammyShittu — COMMENTED** [review]
Hi @Estherstarr,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

### Did you forget to push your changes? There appear to be no new commits reflected any previous change requested earlier. Kindly cross-check and resubmit.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by Lordkaito — APPROVED** [review]
Hi @Estherstarr, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**